Miniscript-Template-000-Timelocks in Templates.mediawiki

```diff
@@ -0,0 +1,81 @@
+
+Should an output descriptor be lost or only partially recovered for any reason, minimizing the overall search space for expected timestamp values can expedite recovery. Although Miniscript supports any valid block height or epoch timestamp for absolute timelocks, the purpose of these proposed templates is to constrain the inherent open-endedness of Miniscript. This aims to foster user-friendly best practices in software wallets and signing devices.
+
+=Block height based Timelocks:=
+
+==Absolute block height based Timelocks==
+
+Proposal: Absolute block height based timelocks should be a multiple of 100, always ending in `00`.
+
+Examples of valid Block Height Absolute Timelocks
+
+*after(1000)
+*after(50700)
+*after(82800)
+*after(615000)
+
+
+== Relative block height based Timelocks ==
+Proposal: Relative block height timelocks should also be multiple of 100, with an exception for the maximum possible value supported, `65,536` (2^16) blocks.
+
+
+*older(100)
+*older(1500)
+*older(65536)
+
+
+=Epoch timestamp based Timelocks:=
+
+==Absolute Epochtime based Timelocks==
+
+*Since the goal of an epoch timestamp timelock is to align events with wall time instead of block time, epoch timestamps used should be of signifigance. Since there is a much larger searchspace with epoch timestamp timelocks, I'm proposing Epoch Timestamp based Timelocks should be set to either Noon, or Midnight GMT. 
+*In practice, this means that absolute epoch timelocks should be divisible by either `43,200` (Noon GMT) or `86,400` (Midnight GMT).
+*Currently in 2106 (at epoch timestamp `4294967295`, February 7th 2106 at 6:28:15 am UTC), the 32 bit unsigned integer used for timestamps will overflow. As a precaution no epoch timestamps should be used past 2105. This is not to say using timelocks this far out is advisable, but to ensure funds are forward compatible, epoch timestamps should not be placed after 2105 (`4291704000`).
+
+
+Propsoed Examples of valid Epoch Timestamp Absolute Timelocks
+
+
+*after(1694476800)
+*after(1694520000)
+*after(2234779200)
+*after(1767139200)
+
+
+
+
+==Relative Epochtime based Timelocks==
+
+Proposed Examples of valid Block Height Relative Timelocks
+- Following [https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki BIP68], relative epoch timestamp timelocks can only go as far out as 33,554,431 seconds (1.06 years), as it is constrained by the same units as relative block height timelocks, 65,536 (2^16), where each unit represents 512 seconds of time (65,356 units * 512 seconds = 33,554,432 seconds).
+
+* An Epoch Timestamp is not valid until the network's MTP (Median Time Past) of the past 11 blocks is greater than the epoch timestamp, MTP is defined in: [https://github.com/bitcoin/bips/blob/master/bip-0113.mediawiki BIP113].
+
+* Since the 512 second incrimentor can not align exactly to a 24 hour day, for consistent recovery sake, relative epoch timestamps should be divisible by 100.
```

Comment:
In 4b75129bd282aa8fc81fe4dbd86c28f870b5ad8b:

`incriment` should be `increment` (in two places).

Also, FWIW, 512 is a divisor of *four* days. You could maybe suggest using multiples of 675? Or "675 or 100"? Or 2700 (16 days) which is the LCM :P. Just suggestions.

Reply:
Reflecting on this, I think if someone is going to use an epoch time relative timelock, it should be grounded in something is relatable in epoch time. So 675 to represent which neatly folds into 4 day relative timelock is better. It can be more cleanly represented in software and hardware wallet interfaces, and is a unit of time that people can reason about much easier than 100 units of 14.2 hours.

Additionally, the values for an epoch time relative timelock in`older()`does not start on 0, so it is going to look unintuive at face value for the output descriptor no matter what.